chia/rpc/wallet_rpc_api.py

```diff
@@ -449,6 +449,10 @@ async def get_wallet_balance(self, request: Dict) -> Dict:
         pending_change = await wallet.get_pending_change_balance()
         max_send_amount = await wallet.get_max_send_amount(unspent_records)
 
```

Comment:
let's use unspent_records fetched above to get the removals. (it avoids doing same query multiple times)

Reply:
unconfirmed_removals_for_wallet() gets unspent_records again, it can be changed to have optional arg unspent_records similar to get_max_send_amount for example.